pkg/rlptools/gateway_utils.go

```diff
@@ -213,3 +214,97 @@ func (g GatewayWrapper) Hostnames() []string {
 
 	return hostnames
 }
+
+func RouteHostnames(route *gatewayapiv1alpha2.HTTPRoute) []string {
+	if route == nil {
+		return nil
+	}
+
+	if len(route.Spec.Hostnames) == 0 {
+		return []string{"*"}
+	}
+
+	hosts := make([]string, 0, len(route.Spec.Hostnames))
+
+	for _, hostname := range route.Spec.Hostnames {
+		hosts = append(hosts, string(hostname))
+	}
+
+	return hosts
+}
+
+// RulesFromHTTPRoute computes a list of rules from the HTTPRoute object
+func RulesFromHTTPRoute(route *gatewayapiv1alpha2.HTTPRoute) []apimv1alpha1.Rule {
+	if route == nil {
+		return nil
+	}
+
+	var rules []apimv1alpha1.Rule
+
+	for routeRuleIdx := range route.Spec.Rules {
+		for matchIdx := range route.Spec.Rules[routeRuleIdx].Matches {
+			match := &route.Spec.Rules[routeRuleIdx].Matches[matchIdx]
+
+			rule := apimv1alpha1.Rule{
+				Hosts: RouteHostnames(route),
+			}
+
+			rule.Methods = RouteHTTPMethodToRuleMethod(match.Method)
+			rule.Paths = routePathMatchToRulePath(match.Path)
+
+			if len(rule.Methods) != 0 || len(rule.Paths) != 0 {
+				// Only append rule when there are methods or path rules
+				// a valid rule must include HTTPRoute hostnames as well
+				rule.Hosts = RouteHostnames(route)
+				rules = append(rules, rule)
+			}
+		}
+	}
+
+	// If no rules compiled from the route, at least one rule for the hosts
+	if len(rules) == 0 {
+		rules = []apimv1alpha1.Rule{{Hosts: RouteHostnames(route)}}
+	}
+
+	return rules
+}
+
+// routePathMatchToRulePath converts HTTPRoute pathmatch rule to kuadrant's rule path
+func routePathMatchToRulePath(pathMatch *gatewayapiv1alpha2.HTTPPathMatch) []string {
+	if pathMatch == nil {
+		return nil
+	}
+
+	// Only support for Exact and Prefix match
+	if pathMatch.Type != nil && *pathMatch.Type != gatewayapiv1alpha2.PathMatchPathPrefix &&
+		*pathMatch.Type != gatewayapiv1alpha2.PathMatchExact {
+		return nil
+	}
+
+	suffix := ""
+	if pathMatch.Type == nil {
+		// defaults to path prefix match type
+		suffix = "*"
+	}
+	if *pathMatch.Type == gatewayapiv1alpha2.PathMatchPathPrefix {
+		suffix = "*"
+	}
+	if *pathMatch.Type == gatewayapiv1alpha2.PathMatchExact {
+		suffix = ""
+	}
+
+	val := "/"
+	if pathMatch.Value != nil {
+		val = *pathMatch.Value
+	}
+
+	return []string{val + suffix}
+}
```

Comment:
Yup, `switch` could improve readability:

```go
suffix := ""
switch t := pathMatch.(type){
      case nil:
        sufix = "*"
      case gatewayapiv1alpha2.PathMatchPathPrefix:
        sufix = "*"

...
```

Reply:
That code does not work, `gatewayapiv1alpha2.PathMatchPathPrefix` is not a type, but a value. 

The best I could come up with is

```go
// Exact path match                                                                      
suffix := ""                                                                             
if pathMatch.Type == nil || *pathMatch.Type == gatewayapiv1alpha2.PathMatchPathPrefix {  
    // defaults to path prefix match type                                                
    suffix = "*"                                                                         
} 
```

If we ever implement Regexp path matching, this code needs to be extended with an additional if case.